### PR TITLE
feat: add check for duplicate testcase builds.

### PR DESCRIPTION
**Pull request by nanjingfm**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [ ] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [ ] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [ ] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [ ] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: .build/build.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@ spec:
           value:
             - passed-tests-rate=100
         - name: tool-image
-          value: registry.alauda.cn:60080/devops/builder-go:1.21
+          value: registry.alauda.cn:60080/devops/builder-go-121:latest
```
Comment (nanjingfm) [inline .build/build.yaml:92]: 稍后修改 hack 仓库

Comment (nanjingfm) [reply .build/build.yaml:92]: 只有 latest tag 才会自动更新镜像，（参考红帽的方案）

**Comment by nanjingfm** [comment]
downstream-test 失败是应为 pvc 满了

```shell
pattern ./...: write /go/pkg/buildcache/02/026482a3a2de2ecc7f0d1f9213c0c1cfa243191d0700c5bbb75a9ca5f4602a34-d: disk quota exceeded
```

File: testing/framework/conformance/testcase.go [hunk 2]
```diff
@@ -45,6 +48,10 @@ func (t *testCase) AddLabels(labels ...Labels) *testCase {
 
 // Build construct a `CaseSetFactory` with the `caseRegister`
 func (t *testCase) Build(caseRegister func(ctx context.Context)) CaseSetFactory {
+	if t.buildFlag {
+		panic("each testcase can only be built once.")
```
Comment (nanjingfm) [inline testing/framework/conformance/testcase.go:52]: panic 可以在开发阶段提前发现问题

File: go.mod [hunk 3]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/HdrHistogram/hdrhistogram-go v1.1.0 // indirect
 	github.com/blendle/zapdriver v1.3.1
 	github.com/caarlos0/env/v6 v6.6.2
-	github.com/cloudevents/sdk-go/v2 v2.14.0
+	github.com/cloudevents/sdk-go/v2 v2.15.2
```
Comment (nanjingfm) [inline go.mod:9]: 升级版本，修复漏洞

**Review by l-qing — APPROVED** [review]

**Pull request merged**